Feature/blog component

kinwits opened the pull request.

kinwits committed:
Blog component

kinwits committed:
Merge dev branch into feature/blog-component - resolved conflicts in Home.jsx and home.routes.jsx

kinwits committed:
Blog component

SaiPraneeth-Kinwits commented on src/styles/theme.css:
```diff
@@ -48,7 +48,7 @@
   name: "dark";
   default: false;
-  prefersdark: false;
+  prefersdark: true;
```
May I know why this was changed to `true`?

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+      <PageHeader title="Blog" />
+
+      <section className="bg-gradient-to-br from-base-100 via-base-100 to-primary/5 py-20">
+        <div className="section-container">
```
add `section-padding` classname here and remove `py-20` on section tag.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+    <main className="pt-16">
+      <PageHeader title="Blog" />
+
+      <section className="bg-gradient-to-br from-base-100 via-base-100 to-primary/5 py-20">
```
I think, `bg-gradient-to-br from-base-100 via-base-100 to-primary/5` not required here.

SaiPraneeth-Kinwits commented:
Kindly run `yarn run lint:fix` before committing changes to fix the Eslint issues.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+import { Link } from "@tanstack/react-router";
+import PageHeader from "@/components/PageHeader";
+
+const allBlogPosts = [
```
Separate static data to a different file and import in this file.
Do this where ever applicable.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+    }
+  }, []);
+
+  const filteredPosts = allBlogPosts.filter((post) => {
```
We can create a `custom hook` for filtering data and use it here and in `BlogPost.jsx` as well.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+              <span className="w-2 h-2 bg-primary rounded-full"></span>
+              Latest Financial Insights
+            </div>
+            <h2 className="text-4xl md:text-5xl font-bold mb-6 bg-gradient-to-r from-base-content to-base-content/80 bg-clip-text text-transparent">
```
to maintain consistency lets remove `text-4xl md:text-5xl`.
Let h2 handle the text sizing. If needed use `h1` else modify h2 values in `typography.css` file

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+                        {/* Meta */}
+                        <div className="flex items-center gap-4 text-sm text-base-content/60 mb-4">
+                          <div className="flex items-center gap-2">
+                            <div className="w-8 h-8 rounded-full bg-primary/20 flex items-center justify-center">
```
use `flex-center` utility classname instead of `flex items-center justify-center`

Apply `flex-center` utility where ever applicable.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+                        </div>
+
+                        {/* Title */}
+                        <h3 className="text-xl font-bold mb-4 text-base-content group-hover:text-primary transition-colors duration-300 line-clamp-2">
```
let h3 handle the sizing, remove `text-xl`

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+                          <span className="text-primary hover:text-primary-focus font-semibold text-sm group-hover:translate-x-2 transition-all duration-300 flex items-center gap-2">
+                            Read More
+                            <svg
+                              className="w-4 h-4 group-hover:translate-x-1 transition-transform duration-300"
```
instead of hardcoded svg we can use icon from `lucide-react` or `heroIcons` library.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+              {filteredPosts.length === 0 && (
+                <div className="text-center py-16">
+                  <div className="w-24 h-24 mx-auto mb-6 bg-base-200 rounded-full flex items-center justify-center">
+                    <svg
```
instead of hardcoded svg we can use icon from `lucide-react` or `heroIcons library`.

Apply the same where ever applicable.

SaiPraneeth-Kinwits commented on src/pages/Blog.jsx:
```diff
@@ -0,0 +1,320 @@
+                            />
+                          </div>
+                          <div className="flex-1 min-w-0">
+                            <h4 className="font-semibold text-base-content text-sm line-clamp-2 group-hover:text-primary transition-colors">
```
`text-sm` on h4?

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,816 @@
+    <main className="pt-16 bg-base-100 dark:bg-gray-900 min-h-screen">
+      <PageHeader title={post.title} />
+
+      <article className="py-20 bg-base-100 dark:bg-gray-900">
```
why `dark:bg-gray-900`?

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,816 @@
+      <PageHeader title={post.title} />
+
+      <article className="py-20 bg-base-100 dark:bg-gray-900">
+        <div className="section-container">
```
add section-padding classname here and remove py-20 on section tag.

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,816 @@
+                </p>
+
+                <div className="grid grid-cols-1 md:grid-cols-2 gap-8 mb-8">
+                  <div className="flex gap-4">
```
instead of writing 4 div for each item, we can loop through and use one div to render content.

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,816 @@
+                    ></textarea>
+                  </div>
+                  <div className="flex items-center gap-4">
+                    <button type="submit" className="btn btn-primary">
```
maintain `button` height consistency.

SaiPraneeth-Kinwits commented on src/components/PageHeader.jsx:
```diff
@@ -21,34 +21,34 @@ const PageHeader = ({ title, customName = false }) => {
 
   return (
-    <section className="text-light relative h-64 w-full bg-gray-900">
+    <section className="text-white relative h-64 w-full bg-gray-900 dark:bg-gray-800">
```
why changes in this file?

kinwits committed:
Fix all PR review comments and improve blog components

- Created useBlogFilter custom hook for reusable filtering logic
- Fixed ArrowRight import errors in Blog.jsx and BlogPost.jsx
- Replaced hardcoded SVGs with lucide-react icons
- Removed redundant dark mode classes and text sizing overrides
- Added section-padding utility classes and removed py-20
- Refactored repeated strategy divs into dynamic loop
- Maintained consistent button sizing with btn-lg
- Fixed navigation issues in blog post cards
- Updated typography to use semantic sizing system

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,1031 @@
+                    </p>
+
+                    <div className="mb-8 grid grid-cols-1 gap-8 md:grid-cols-2">
+                      {[
```
can we have custom data in a separate variable?

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,1031 @@
+                    </h3>
+                    <div className="flex flex-wrap gap-4">
+                      <button className="flex h-12 w-12 items-center justify-center rounded-xl bg-blue-100 text-blue-600 transition-colors hover:bg-blue-200">
+                        <svg
```
instead of svg's can we use icons imported from lucide-react or heroIcons library.

SaiPraneeth-Kinwits commented on src/pages/BlogPost.jsx:
```diff
@@ -0,0 +1,1031 @@
+                  </h3>
+                  <div className="flex flex-wrap gap-2">
+                    {[
+                      "Investment",
```
same here, assign them to a variable and use it here
